jdaviz/configs/imviz/helper.py

```diff
@@ -286,6 +287,130 @@ def zoom(self, val):
             raise ValueError(f"zoom only accepts int or float but got '{val}'")
         self.zoom_level = self.zoom_level * val
 
+    @property
+    def colormap_options(self):
+        """List of colormap names."""
+        return sorted(member[1].name for member in colormaps.members)
+
+    def set_colormap(self, cmap):
+        """Set colormap to the given colormap name.
+
+        Parameters
+        ----------
+        cmap : str
+            Colormap name. Possible values can be obtained from
+            :meth:`colormap_options`.
+
+        Raises
+        ------
+        ValueError
+            Invalid colormap name.
+
+        """
+        cm = None
+        for member in colormaps.members:
+            if member[1].name == cmap:
+                cm = member[1]
+                break
+
+        if cm is None:
+            raise ValueError(f"Invalid colormap '{cmap}', must be one of {self.colormap_options}")
+
+        viewer = self.app.get_viewer("viewer-1")
+        i_top = get_top_layer_index(viewer)
+        viewer.state.layers[i_top].cmap = cm
+
+    @property
+    def stretch_options(self):
+        """List of all available options for image stretching.
+
+        Their ``astropy.visualization`` counterparts are also accepted, as follows:
+
+        * ``'arcsinh'``: ``astropy.visualization.AsinhStretch``
+        * ``'linear'``: ``astropy.visualization.LinearStretch``
+        * ``'log'``: ``astropy.visualization.LogStretch``
+        * ``'sqrt'``: ``astropy.visualization.SqrtStretch``
+
+        """
+        # TODO: Is there a better way to access this in Glue? See glue/viewers/image/state.py
+        return ['arcsinh', 'linear', 'log', 'sqrt']
+
+    @property
+    def stretch(self):
+        """The image stretching algorithm in use."""
+        viewer = self.app.get_viewer("viewer-1")
+        i_top = get_top_layer_index(viewer)
+        return viewer.state.layers[i_top].stretch
+
+    @stretch.setter
+    def stretch(self, val):
+        valid_vals = self.stretch_options
+
+        if isinstance(val, type):  # is a class
+            # Translate astropy.visualization
+            from astropy.visualization import AsinhStretch, LinearStretch, LogStretch, SqrtStretch
+            if issubclass(val, AsinhStretch):
```

Comment:
Would it be possible to allow the user to put in a string value (i.e. 'log', 'linear', 'asinh', 'sqrt') and have us "autocorrect" behind the scenes to the right `astropy.visualization` class? I feel like that would be a simple `ease-of-use` addon.

Reply:
D.U.E. ... nevermind

Reply:
Actually, it is the other way around. Glue takes string and we're trying to translate astropy object to the string that Glue understands, as @eteq wants some astropy object support.

Re: D.U.E... I can't tell if you are trying to work your name into the software or I am missing some new CS terminologies here. 😸

Reply:
Sorry. "Dumb User Error." I was a dummy. Made a mistake. An oopsie. Please ignore my incompetence ;)